http-clients/netty-nio-client/src/main/java/software/amazon/awssdk/http/nio/netty/NettyNioAsyncHttpClient.java

```diff
@@ -145,8 +153,12 @@ protected ChannelPool newPool(URI key) {
                         .channelFactory(sdkEventLoopGroup.channelFactory())
                         .option(ChannelOption.CONNECT_TIMEOUT_MILLIS, configuration.connectTimeoutMillis())
                         // TODO run some performance tests with and without this.
-                        .option(ChannelOption.TCP_NODELAY, true)
                         .remoteAddress(key.getHost(), key.getPort());
+
+                for (Map.Entry<ChannelOption, Object> entry : sdkChannelOptions.channelOptions().entrySet()) {
+                    bootstrap.option(entry.getKey(), entry.getValue());
+                }
```

Comment:
nit: line 157 - line 160 can be replaced with `sdkChannelOptions.channelOptions().forEach(bootstrap::option);` :)